src/Tests.IntegrationTests/ClassifyMessageTests.cs

```diff
@@ -0,0 +1,86 @@
+using Microsoft.Extensions.Configuration;
+using System.Text.Json;
+using Tests.IntegrationTests.Models;
+
+namespace Tests.IntegrationTests
+{
+    public class ClassifyMessageTests
+    {
+        private readonly IConfiguration _configuration;
+
+        public ClassifyMessageTests()
+        {
+            _configuration = new ConfigurationBuilder()
+                .AddJsonFile(@"appsettings.json", false, false)
+                .Build();
+        }
+
+        [Fact(Timeout = 120000)]
+        public async Task GivenValidMessageReceivesValidResponse()
+        {
+            // Arrange
+            using var httpClient = new HttpClient();
+            var bot1Url = _configuration["Bot1Url"];
+            var chatsUri = new Uri($"{bot1Url}/client-api/chats");
+
+            // Act
+            // 1 Create Chat
+            var createdChat = await Request<Chat>(httpClient, Verb.Post, chatsUri).ConfigureAwait(false) as Chat;
+
+            // 2 Create Message
+            var chatMessageUri = new Uri($"{bot1Url}/client-api/chats/{createdChat.Id}/messages");
+            using var content = new StringContent("i want to register my child at school");
+            var createdChatMessage = await Request<ChatMessage>(httpClient, Verb.Post, chatMessageUri, content).ConfigureAwait(false) as ChatMessage;
+
+            // 3 Retry until we have 2 messages in the chat that this test created
+            Chat resultChat = null;
+            bool breakLoop = false;
+            while (!breakLoop)
+            {
+                var chats = await Request<List<Chat>>(httpClient, Verb.Get, chatsUri).ConfigureAwait(false) as List<Chat>;
+                resultChat = chats.First(c => c.Id == createdChat.Id);
+                if (resultChat.Messages.Count < 2)
+                {
+                    await Task.Delay(new TimeSpan(0, 0, 10)).ConfigureAwait(false);
+                }
+                else
+                {
+                    breakLoop = true;
+                }
+            }
```

Comment:
I would back your suggestion during standup this morning and push for using Polly for this.   The problem with this code is that if the chat doesn't update to 2 messages this code wait with 10 second intervals forever and won't timeout.

If you want to stick with this code but add a timeout, I would suggest something like this:
```suggestion
            Func<Task<Chat> fetchChatAsync = () => 
            {
                bool breakLoop = false;
                while (!breakLoop)
                {
                    var chats = await Request<List<Chat>>(httpClient, Verb.Get, chatsUri).ConfigureAwait(false) as List<Chat>;
                    var resultChat = chats.First(c => c.Id == createdChat.Id);
                    breakLoop = resultChat.Messages.Count < 2;
                    if (breakLoop)
                    {
                        return resultChat;
                    }
                    else
                    {
                        await Task.Delay(new TimeSpan(0, 0, 10)).ConfigureAwait(false);
                    }
                }
            };

            var resultChatTask = fetchChatAsync();

            var task = await Task.WhenAny(resultChatTask, Task.Delay(TimeSpan.FromSeconds(60));

            Assert.Equal(task, resultChatTask); // Will fail if the Delay task returns in 60 seconds.
            
            var resultChat = await resultChatTask;
```

Hope that makes sense.

Reply:
I did look at Polly, but I concluded that it is a bit OTT for what is needed here as it is more about retrying for exceptions than because the result is not what was expected, so Polly did not really match what I wanted. I'm sure it could be bent to do what I want but using a whole library for just a few lines of basic C# seemed excessive to me. There is already a timeout set for the test on `Fact(Timeout = 120000)]` so I don't think this refactoring is required?